dvc/repo/experiments/executor/base.py

```diff
@@ -262,16 +260,6 @@ def _from_stash_entry(
         )
         return executor
 
-    @staticmethod
-    def hash_exp(stages: Iterable["PipelineStage"]) -> str:
```

Comment:
I think it might be worth leaving the existing hashing behavior alone. (So this PR would just add the new code needed to generate the random human-readable names (and always set `name` before running the exp which will make us skip the current duplicate detection).

It seems like handling duplicates in the future will mean we still need these hashes, and with the desired behavior change to make `exp apply` work without requiring the matching baseline, I think we could change the exp ref namespace to look something like
`refs/exps/<exp_hash>/...` which would make checking for duplicates in the future straightforward

Reply:
Added it back